lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1121,6 +1125,26 @@ sub _create_bam_flagstat {
     return 1;
 }
 
+sub create_bam_header {
+    my $self = shift;
+
+    if (-s $self->bam_header_path) {
+        return 1;
+    }
+
+    my $lock = $self->get_bam_lock;
```

Comment:
I was under the (probably mistaken) impression that locks are already released when they go out of scope. This seems clean to me, though, so I will include it where it is useful.